scripts/project-setup-scripts/google-project-owner.yaml

```diff
@@ -0,0 +1,1683 @@
+- servicenetworking.services.get
+- serviceusage.operations.get
+- serviceusage.operations.list
+- serviceusage.quotas.get
+- serviceusage.services.get
+- serviceusage.services.list
+- source.repos.get
+- source.repos.getIamPolicy
+- source.repos.list
+- spanner.backupOperations.get
+- spanner.backupOperations.list
+- spanner.backups.get
+- spanner.backups.getIamPolicy
+- spanner.backups.list
+- spanner.databaseOperations.get
+- spanner.databaseOperations.list
+- spanner.databases.beginReadOnlyTransaction
+- spanner.databases.get
+- spanner.databases.getDdl
+- spanner.databases.getIamPolicy
+- spanner.databases.list
+- spanner.databases.partitionQuery
+- spanner.databases.partitionRead
+- spanner.databases.read
+- spanner.databases.select
+- spanner.instanceConfigs.get
+- spanner.instanceConfigs.list
+- spanner.instanceOperations.get
+- spanner.instanceOperations.list
+- spanner.instances.get
+- spanner.instances.getIamPolicy
+- spanner.instances.list
+- spanner.sessions.create
+- spanner.sessions.delete
+- spanner.sessions.get
+- spanner.sessions.list
+- stackdriver.projects.get
+- storage.buckets.list
+- storage.hmacKeys.get
+- storage.hmacKeys.list
+- storagetransfer.jobs.get
+- storagetransfer.jobs.list
+- storagetransfer.operations.get
+- storagetransfer.operations.list
+- storagetransfer.projects.getServiceAccount
+- subscribewithgoogledeveloper.tools.get
+- tpu.acceleratortypes.get
+- tpu.acceleratortypes.list
+- tpu.locations.get
+- tpu.locations.list
+- tpu.nodes.get
+- tpu.nodes.list
+- tpu.operations.get
+- tpu.operations.list
+- tpu.tensorflowversions.get
+- tpu.tensorflowversions.list
+- trafficdirector.networks.getConfigs
+- transcoder.jobTemplates.get
+- transcoder.jobTemplates.list
+- transcoder.jobs.get
+- transcoder.jobs.list
+- vmmigration.deployments.get
+- vmmigration.deployments.list
+- vmwareengine.services.view
+- vpcaccess.connectors.get
+- vpcaccess.connectors.list
+- vpcaccess.connectors.use
+- vpcaccess.locations.list
+- vpcaccess.operations.get
+- vpcaccess.operations.list
+- workflows.executions.get
+- workflows.executions.list
+- workflows.locations.get
+- workflows.locations.list
+- workflows.operations.get
+- workflows.operations.list
+- workflows.workflows.get
+- workflows.workflows.getIamPolicy
+- workflows.workflows.list
+# </viewer> End of permissions included as part of roles/viewer
+# Add permissions for the other roles we need:
+##roles/bigquery.jobUser (updated 2/10/2021)
+- bigquery.jobs.create
+- resourcemanager.projects.get
+- resourcemanager.projects.list
+##roles/logging.logWriter (updated 2/10/2021)
+- logging.logEntries.create
+##roles/monitoring.metricWriter (updated 2/10/2021)
+- monitoring.metricDescriptors.create
+- monitoring.metricDescriptors.get
+- monitoring.metricDescriptors.list
+- monitoring.monitoredResourceDescriptors.get
+- monitoring.monitoredResourceDescriptors.list
+- monitoring.timeSeries.create
+##RequesterPays custom role (updated 2/10/2021)
+- serviceusage.services.use
```

Comment:
There's a LOT getting set here.  Yes it's just read access for a lot of it, but still.  I think Bernick had made a request at one point to review permissions on things to make sure we're truly granting only permissions we need.  In other words, start with 0 permissions and only grant what's needed, as opposed to grant EVERYTHING and then remove permissions to things you don't want people getting at.  It seems like we could go through and bulk remove permissions for things like, `ml`, `gameservices`, `aiplatform`, etc.  If we need to add things in the future, it will be easy to do now that we have custom roles.

This seems like something InfoSec will care about too.  I feel like it should be pretty easy for us to chop hundreds of permissions away that we know we don't need these days.

Reply:
I've pruned this list in several passes, getting progressively more aggressive each pass. See commit history for details!